fix(examples): add missing readme for Pagination examples (#209)
fivetran/fivetran_connector_sdk: examples/common_patterns_for_connectors/pagination/offset_based/README.md

- Supports ordering by an update key `updatedAt`.

This example is intended for learning purposes and uses the [fivetran-api-playground](https://pypi.org/project/fivetran-api-playground/) package to mock the API responses locally. It is not meant for production use.

## Requirements
- [Supported Python versions](https://github.com/fivetran/fivetran_connector_sdk/blob/main/README.md#requirements)   
- Operating system:
  - Windows: 10 or later (64-bit only)
  - macOS: 13 (Ventura) or later (Apple Silicon [arm64] or Intel [x86_64])
  - Linux: Distributions such as Ubuntu 20.04 or later, Debian 10 or later, or Amazon Linux 2 or later (arm64 or x86_64)

## Getting started
Refer to the [Setup Guide](https://fivetran.com/docs/connectors/connector-sdk/setup-guide) to get started.

## Features
- Demonstrates classic offset-based pagination with `limit` and `offset`.
- Filters results using an `updated_since` timestamp for incremental syncs.
- Supports safe and resumable state checkpointing using `op.checkpoint()`.
- Updates the replication cursor based on `updatedAt` timestamp from the latest record.

## Configuration file
This example does not require a configuration file.

> In production, configuration.json might contain API tokens, initial cursors, or filters to narrow down API results.

Reviewer: ```suggestion
For production connectors, `configuration.json` might contain API tokens, initial cursors, or filters to narrow down API results.
```

## Authentication
This connector does not use authentication.

In real-world scenarios, modify `get_api_response()` to add `Authorization` headers or include API keys in query parameters.

## Pagination
Pagination is handled using the following parameters:
- `offset`: Current position in the record set.
- `limit`: Number of records per page (e.g., 50).
- `total`: Total number of records available (from API response).

Pagination logic:
- Continues requesting pages while `(offset + page_size) < total`.
- Advances the offset by the number of records returned.
- Ends when the full dataset has been consumed.

## Data handling
- Starts sync with` updated_since = '0001-01-01T00:00:00Z'` if no state is present.
- Each record is passed to `op.upsert()` with destination table `USER`.
- The state is updated with the latest `updatedAt` value from each page.
- After each page, `op.checkpoint()` saves progress to ensure resumability.

## Error handling
- API errors are raised via `response.raise_for_status()`.
- Handles empty pages or partial results gracefully.
- Logs the start of each page and first record ID for traceability.

## Tables Created

> The connector creates one table:

Reviewer: ```suggestion
The connector creates the `USER` table:
```
